src/UnitTests/LanguageServerClient/UnitTests.LanguageServerClient/PythonLanguageServerDotNetCore.cs

```diff
@@ -0,0 +1,244 @@
+                        "**/node_modules",
+                        "**/bower_components",
+                        "**/.git",
+                        "**/.svn",
+                        "**/.hg",
+                        "**/CVS",
+                        "**/.DS_Store",
+                        "**/.git/objects/**",
+                        "**/.git/subtree-cache/**",
+                        "**/node_modules/*/**",
+                        ".vscode/*.py",
+                        "**/site-packages/**/*.py"
+                    },
+                rootPathOverride = rootPath
+            };
+        }
+
+        private static ProcessStartInfo CreateStartInfo(string folderPath) {
+            var serverDllFilePath = Path.Combine(folderPath, DllName);
+            var serverExeFilePath = Path.Combine(folderPath, ExeName);
+
+            if (File.Exists(serverExeFilePath)) {
+                return new ProcessStartInfo {
+                    FileName = serverExeFilePath,
+                    WorkingDirectory = folderPath,
+                    RedirectStandardInput = true,
+                    RedirectStandardOutput = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true
+                };
+            } else if (File.Exists(serverDllFilePath)) {
+                return new ProcessStartInfo {
+                    FileName = DotNetExeFilePath,
+                    WorkingDirectory = folderPath,
+                    Arguments = '"' + serverDllFilePath + '"',
+                    RedirectStandardInput = true,
+                    RedirectStandardOutput = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true
+                };
+            } else {
+                Debug.Fail("Could not find language server exe or dll");
+                throw new FileNotFoundException("Could not find language server exe or dll", serverDllFilePath);
+            }
+        }
+
+        private string GetServerLocation() {
+            // For testing, allow to specify via env var the location
+            // of an extracted language server
+            var lsExtractedFolderPath = Environment.GetEnvironmentVariable("PTVS_DOTNETCORE_SERVER_LOCATION");
+            if (Directory.Exists(lsExtractedFolderPath)) {
+                return lsExtractedFolderPath;
+            }
+
+            if (!_joinableTaskContext.IsOnMainThread) {
+                throw new InvalidOperationException("Must be called from main thread.");
+            }
+
+            //var shell = _site.GetService<SVsShell, IVsShell>();
+            //shell.GetProperty((int)__VSSPROPID4.VSSPROPID_LocalAppDataDir, out object localAppDataFolderPath);
+
+            //lsExtractedFolderPath = Path.Combine((string)localAppDataFolderPath, "PythonLanguageServer");
+            //if (!Directory.Exists(lsExtractedFolderPath)) {
+            //    ExtractToFolder(lsExtractedFolderPath);
+            //}
```

Comment:
commented out code related to VS.. currently I'm not extracting the server from a zip folder. I'm just using the build output folder in MPLS